ui/components/ui/survey-toast/survey-toast.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState, useContext } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import fetchWithCache from '../../../../shared/lib/fetch-with-cache';
+import { DAY } from '../../../../shared/constants/time';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import {
+  getSelectedInternalAccount,
+  getLastViewedUserSurvey,
+  getUseExternalServices,
+  getParticipateInMetaMetrics,
+} from '../../../selectors';
+import { setLastViewedUserSurvey } from '../../../store/actions';
+import { Toast } from '../../multichain';
+
+type Survey = {
+  url: string;
+  description: string;
+  cta: string;
+  surveyId: number;
+};
+
+export function SurveyToast() {
+  const [survey, setSurvey] = useState<Survey | null>(null);
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const lastViewedUserSurvey = useSelector(getLastViewedUserSurvey);
+  const participateInMetaMetrics = useSelector(getParticipateInMetaMetrics);
+  const basicFunctionality = useSelector(getUseExternalServices);
+  const internalAccount = useSelector(getSelectedInternalAccount);
+
+  useEffect(() => {
+    if (!basicFunctionality || !internalAccount?.address) {
+      return;
+    }
+
+    const surveyId = 1;
+    const surveyUrl = `https://accounts.dev-api.cx.metamask.io/v1/users/${internalAccount.address}/surveys?surveyId=${surveyId}`;
+
+    const fetchSurvey = async () => {
+      try {
+        const response = await fetchWithCache({
+          url: surveyUrl,
+          fetchOptions: {
+            method: 'GET',
+            headers: {
+              'x-metamask-clientproduct': 'metamask-extension',
+            },
+          },
+          functionName: 'fetchSurveys',
+          cacheOptions: { cacheRefreshTime: DAY * 7 },
+        });
+
+        const _survey: Survey = response?.surveys?.[0];
+
+        if (
+          response.surveys.length === 0 ||
+          !_survey ||
+          _survey.surveyId <= lastViewedUserSurvey
+        ) {
+          return;
+        }
+
+        setSurvey(_survey);
+      } catch (error) {
+        console.error('Failed to fetch survey:', error);
+      }
+    };
+
+    fetchSurvey();
+  }, [
+    internalAccount?.address,
+    lastViewedUserSurvey,
+    basicFunctionality,
+    dispatch,
+  ]);
+
+  function handleActionClick() {
+    if (!survey) {
```

Comment:
this check is redundant. you can most likely remove it

Reply:
I get a TS error and I am not allowed to use the non-null assertion (!) e.g. survey!.url
The TS issue is happening because I am coding this as part of a function.